### PR TITLE
Module changes for Identity & Auth

### PR DESCRIPTION
**Pull request by gosar**
## Motivation and Context

As part of Smithy Reference Architecture, we are defining new Identity and Auth interfaces that will reside in new modules.

## Modifications

This change adds new modules and makes existing modules will depend on these new modules for backwards-compatibility. The actual code changes will be done in separate PRs. This change preps the module changes only. To allow the build to succeed, Placeholder.java classes are added to make the module jar non-empty and also maven-dependency-plugin has been temporarily disabled.

## Testing

`./mvnw clean install`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md) document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] I have added a changelog entry. Adding a new entry must be accomplished by running the `scripts/new-change` script and following the instructions. Commit the new file created by the script in `.changes/next-release` with your changes.
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: core/http-auth-aws-crt/pom.xml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    </parent>
+
+    <artifactId>http-auth-aws-crt</artifactId>
+    <name>AWS Java SDK :: HTTP Auth AWS CRT</name>
+    <description>
+        The AWS SDK for Java - Auth module holds the AWS Common Runtime based implementations that are used for
+        authentication with HTTP services
+    </description>
+    <!-- TODO: this is not present in all pom.xml files. Is there a reason, or just a miss -->
+    <url>https://aws.amazon.com/sdkforjava</url>
```
Comment (millems) [inline core/http-auth-aws-crt/pom.xml:35]: Probably just a miss.. Unless it's inherited from the parent?

Comment (gosar) [reply core/http-auth-aws-crt/pom.xml:35]: Hmm, doesn't seem to be inherited. https://github.com/aws/aws-sdk-java-v2/blob/master/http-client-spi/pom.xml doesn't have it and https://search.maven.org/artifact/software.amazon.awssdk/http-client-spi doesn't have _Home page_ link. Compared to https://search.maven.org/artifact/software.amazon.awssdk/auth which does.

Comment (gosar) [reply core/http-auth-aws-crt/pom.xml]: removed these TODOs and another one for HTTP v/s Http as we use HTTP in other modules already.

Comment (millems) [reply core/http-auth-aws-crt/pom.xml]: Must just be a miss, then!

**Pull request merged**